deep_qa/training/trainer.py

```diff
@@ -319,7 +319,7 @@ def train(self):
         # so you may have left it above zero on accident.
         if self.validation_arrays is not None:
             kwargs['validation_data'] = self.validation_arrays
-        elif self.validation_split > 0.0:
+        elif self.validation_split > 0.0 and isinstance(self.training_arrays, tuple):
```

Comment:
Instead of these tuple checks everywhere in `Trainer`, can we use `self.use_data_generator == True` instead? It would be easier to follow the logic.